ee/server/service/vpp.go

```diff
@@ -26,32 +24,17 @@ func (svc *Service) getVPPToken(ctx context.Context) (string, error) {
 		return "", ctxerr.Wrap(ctx, err, "fetching vpp token")
 	}
 
-	var vppTokenData fleet.VPPTokenData
-	if err := json.Unmarshal(configMap[fleet.MDMAssetVPPTokenDeprecated].Value, &vppTokenData); err != nil {
-		return "", ctxerr.Wrap(ctx, err, "unmarshaling VPP token data")
-	}
-
-	vppTokenRawBytes, err := base64.StdEncoding.DecodeString(vppTokenData.Token)
-	if err != nil {
-		return "", ctxerr.Wrap(ctx, err, "decoding raw vpp token data")
-	}
-
-	var vppTokenRaw fleet.VPPTokenRaw
-
-	if err := json.Unmarshal(vppTokenRawBytes, &vppTokenRaw); err != nil {
-		return "", ctxerr.Wrap(ctx, err, "unmarshaling raw vpp token data")
-	}
-
-	exp, err := time.Parse("2006-01-02T15:04:05Z0700", vppTokenRaw.ExpDate)
+	vppTok := &fleet.VPPToken{Token: configMap[fleet.MDMAssetVPPTokenDeprecated].Value}
+	appleTok, _, err := vppTok.ExtractToken()
```

Comment:
I'm not crazy about this design, but I tried a function that took only the stored `[]byte` token and return a `fleet.VPPToken` was weird too as it has no proper way to return the actual raw token that needs to be passed to Apple's API, unless we add a floating field in the VPPToken struct that is only set after a certain function is called... I opted for this approach^  instead.